src/traefik.py

```diff
@@ -0,0 +1,600 @@
+
+        # Must restart traefik after refreshing certs, otherwise:
+        # - newly added certs will not be loaded and traefik will keep erroring-out with "signed by
+        #   unknown authority".
+        # - old certs will be kept active.
+        self.restart()
+
+    def _update_static_configuration(self, tcp_entrypoints: Dict[str, int]):
+        """Generate and push Traefik's static config yaml file."""
+        logger.debug(f"Statically configuring traefik with tcp entrypoints: {tcp_entrypoints}.")
+
+        # TODO Disable static config with telemetry and check new version
+        raw_config = {
+            "log": {
+                "level": "DEBUG",
+            },
+            "entryPoints": {
+                "diagnostics": {"address": f":{_DIAGNOSTICS_PORT}"},
+                "web": {"address": f":{_PORT}"},
+                "websecure": {"address": f":{_TLS_PORT}"},
+                **{
+                    tcp_entrypoint_name: {"address": f":{port}"}
+                    for tcp_entrypoint_name, port in tcp_entrypoints.items()
+                },
+            },
+            # We always start the Prometheus endpoint for simplicity
+            # TODO: Generate this file in the dynamic configuration folder when the
+            #  metrics-endpoint relation is established?
+            "metrics": {
+                "prometheus": {
+                    "addRoutersLabels": True,
+                    "addServicesLabels": True,
+                    "entryPoint": "diagnostics",
+                }
+            },
+            "ping": {"entryPoint": "diagnostics"},
+            "providers": {
+                "file": {
+                    "directory": DYNAMIC_CONFIG_DIR,
+                    "watch": True,
+                }
+            },
+        }
+        config = yaml.safe_dump(raw_config)
+
+        # TODO Use the Traefik user and group?
+        self._container.push(STATIC_CONFIG_PATH, config, make_dirs=True)
+
+    # wokeignore:rule=master
+    # ref: https://doc.traefik.io/traefik/master/observability/tracing/opentelemetry/
+    def update_tracing_configuration(self, endpoint: str, grpc: bool):
+        """Push yaml config with opentelemetry configuration."""
+        config = yaml.safe_dump(
+            {
+                "tracing": {
+                    "openTelemetry": {
+                        "address": endpoint,
+                        **({"grpc": {}} if grpc else {}),
+                        # todo: we have an option to use CA or to use CERT+KEY (available with mtls) authentication.
+                        #  when we have mTLS, consider this again.
+                        **({"ca": CA_CERT_PATH} if self._tls_enabled else {"insecure": True}),
+                    }
+                }
+            }
+        )
+        logger.debug(f"dumping tracing config to {DYNAMIC_TRACING_PATH}")
+
+        self._container.push(DYNAMIC_TRACING_PATH, config, make_dirs=True)
+
+    def get_per_unit_http_config(
```

Comment:
Should this and related methods be moved to a separate class?

Reply:
Will externalize config in a follow-up pr